pkg/proc/proc.go

```diff
@@ -845,3 +847,17 @@ func setAsyncPreemptOff(p *Target, v int64) {
 	err = scope.setValue(asyncpreemptoffv, newConstant(constant.MakeInt64(v), scope.Mem), "")
 	logger.Warnf("could not set asyncpreemptoff %v", err)
 }
+
+// DisableAsyncPreemptEnv returns a process environment (like os.Environ)
+// where asyncpreemptoff is set to 1.
+func DisableAsyncPreemptEnv() []string {
```

Comment:
We already disable async preemption by writing the value we want in the process memory at debug.asyncpreemptoff, why add this method in as well?

Reply:
Because that happens before any runtime code runs, if asyncpreemptoff in GODEBUG is set it will overwrite any change we make (on the other hand if we are attaching the value of GODEBUG will do nothing -- unless we attached very early in the process life).

Reply:
Ack, sorry that's my fault, I misread this patch thinking this was net new, but we were already disabling this in the env, which makes complete sense. Apologies for the noise.